### DISCUSSION TITLE
Filter/Show data by appended attribute or from relationship

### DISCUSSION THREAD

**Discussion by BenOussama180** [rappasoft/laravel-livewire-tables#1065] [Q&A]
so i have this in my link model : 
```
protected $appends = [
        'is_on_order',
    ];

    public function orders()
    {
        return $this->hasMany(Order::class, 'link_id');
    }
    public function getIsOnOrderAttribute()
    {
        return $this->orders->where('user_id', auth()->id())->isEmpty();
    }
```

and this in my LinkTable Component 
```
public function builder(): Builder
    {
        return Link::query()
            ->with('orders')
            ->groupBy('site');
    }
public function columns(): array
    {
        return [
            //some other cols....

            BooleanColumn::make("Purchased Before", "is_on_order")
                ->sortable(),

        ];
    }

```
**the thing is, i can not access the 'is_on_order' attribute to show if the current link has been ordered by a user or not , how can i accesss it ? is there any othher workaround for it by using the orders relationship in my model ? please help**
@rappasoft any idea ? thanks in andvanced !

**Comment by lrljoe** [comment]
I might be getting a bit confused here, so please excuse any ignorance!  But could you not use the where clause within your with() statement to load in the appropriate "on order" bool?

Just a high level example that'd get orders that the user_id matches the Auth::id()
```
return Link::with(['orders' => function (Builder $query) {
    $query->where('user_id', Auth::id());
}]);
```
Or you can use scopes e.g. [Local Scopes](https://laravel.com/docs/10.x/eloquent#local-scopes) to restrict your loads (where applicable)

**Reply by BenOussama180** [reply]
the table is like a market place, i need to show every link, and show which link that has been bought in that purchased before boolean column, but i still cant access the is_on_order attribute

**Reply by lrljoe** [reply]
If it's a hasMany relation, are you imploding /passing to a view the list of values?
